### ISSUE TITLE
reading without a linebreak

### ISSUE THREAD

**Issue by xloem** [mortoray/shelljob#3]
it's be nice to wait until there is data output, and then only read as much as was output, even if there is no linebreak.

for responding to prompts.

**Comment by xloem** [comment]
locally i've actually written a function that takes a dict, and waits to receive its keys (in dict order), and then outputs the values in reply.

**Comment by mortoray** [comment]
Which API call does this relate to?

**Comment by xloem** [comment]
I don't believe there is a specific API call at this time.  It could be added a number of places, and any of them would likely be enough for somebody to work off.  It's a use-case concern that would make the library more usable to more people.

I've moved on to different projects myself at this time.

**Comment by mortoray** [comment]
I haven't considered a lot of interaction in the API design; there's no correct way to write to the external program now either. I mainly meant for reading.

Non-line based reading would also be possible, but may not be easy due to the encoding applied to the stream. It's read in text mode now, that might need to be changed.
